File: java/HelloWorldServer.java
package com.jianchao.thrift;

import com.jianchao.thrift.HelloWorld.Processor;

import org.apache.thrift.server.TServer;
import org.apache.thrift.server.TServer.Args;
import org.apache.thrift.server.TSimpleServer;
import org.apache.thrift.transport.TServerSocket;

import java.net.ServerSocket;

public class HelloWorldServer {
	public static void main(String[] args) throws Exception {
		ServerSocket socket = new ServerSocket(9090);
		TServerSocket serverTransport = new TServerSocket(socket);
		Processor processor = new Processor(new HelloWorldImpl());
                TServer server = new TSimpleServer(new Args(serverTransport).processor(processor));
		System.out.println("Starting thrift server in java...");
		server.serve();
	}
}

File: go/helloworldserver.go
package main

import (
    "fmt"
    "git.apache.org/thrift.git/lib/go/thrift"
    "HelloWorld"
)

func runServer(transportFactory thrift.TTransportFactory, 
	protocolFactory thrift.TProtocolFactory, addr string) error {
	var err error
	transport, err := thrift.NewTServerSocket(addr)
	if err != nil {
		fmt.Println(err)
	}
	fmt.Printf("%T\n", transport)
	handler := NewHelloWorldHandler()
	processor := HelloWorld.NewHelloWorldProcessor(handler)
	server := thrift.NewTSimpleServer4(processor, transport, transportFactory, protocolFactory)
	fmt.Println("Starting thrift server in go...")
	return server.Serve()
}

func main() {
	transportFactory := thrift.NewTTransportFactory()
	protocolFactory := thrift.NewTBinaryProtocolFactoryDefault()
	transportFactory = thrift.NewTBufferedTransportFactory(8192)
	runServer(transportFactory, protocolFactory, "localhost:9090")
}
